packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -458,13 +400,31 @@ class _DefaultPub implements Pub {
     return environment;
   }
 
-  /// Insert the flutter_gen synthetic package into the package configuration file if
-  /// there is an l10n.yaml.
-  Future<void> _updatePackageConfig(File packageConfigFile, Directory generatedDirectory) async {
-    if (!packageConfigFile.existsSync()) {
+  /// Update the package configuration file.
+  ///
+  /// Creates a corresponding `package_config_subset` file that is used by the build
+  /// system to avoid rebuilds caused by an updated pub timestamp.
+  ///
+  /// if [generateSyntheticPackage] is true then insert flutter_gen synthetic
+  /// package into the package configuration.
+  Future<void> _updatePackageConfig(
+    File packageConfigFile,
+    Directory generatedDirectory,
+    bool generateSyntheticPackage,
+  ) async {
+    final PackageConfig packageConfig = await loadPackageConfigWithLogging(packageConfigFile, logger: _logger);
+
+    packageConfigFile.parent
```

Comment:
This is where the subset of package contents are written out, avoids the generated by and generated date fields